clients/admin-ui/src/features/common/custom-fields/hooks.ts

```diff
@@ -0,0 +1,191 @@
+import { useCallback, useMemo } from "react";
+
+import { useFeatures } from "~/features/common/features";
+import { useAlert } from "~/features/common/hooks";
+import {
+  useDeleteCustomFieldMutation,
+  useGetAllAllowListQuery,
+  useGetCustomFieldDefinitionsByResourceTypeQuery,
+  useGetCustomFieldsForResourceQuery,
+  useUpsertCustomFieldMutation,
+} from "~/features/plus/plus.slice";
+import { ResourceTypes } from "~/types/api";
+
+import { filterWithId } from "./helpers";
+import { CustomFieldsFormValues } from "./types";
+
+type UseCustomFieldsOptions = {
+  resourceFidesKey?: string;
+  resourceType: ResourceTypes;
+};
+
+export const useCustomFields = ({
+  resourceFidesKey,
+  resourceType,
+}: UseCustomFieldsOptions) => {
+  const { errorAlert, successAlert } = useAlert();
+  const { flags, plus } = useFeatures();
+  const isEnabled = flags.customFields && plus;
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const queryFidesKey = useMemo(() => resourceFidesKey ?? "", []);
```

Comment:
I mean to put a code comment here explaining that this prevents re-querying for a custom fields that will just 404.